satpy/readers/satpy_cf_nc.py

```diff
@@ -282,6 +282,27 @@ def _coordinate_datasets(self, configured_datasets=None):
             self.fix_modifier_attr(ds_info)
             yield True, ds_info
 
+    def _compare_attr(self, _ds_id_dict, key, data):
+        if key in ['name', 'modifiers']:
+            return True
+        elif key == 'wavelength':
+            if _ds_id_dict[key] != WavelengthRange.from_cf(data.attrs[key]):
+                return False
+        else:
+            if data.attrs[key] != _ds_id_dict[key]:
+                return False
+        return True
+
+    def _dataid_attrs_equal(self, ds_id, data):
+        _ds_id_dict = ds_id.to_dict()
+        for key in _ds_id_dict:
+            try:
+                if not self._compare_attr(_ds_id_dict, key, data):
+                    return False
```

Comment:
```suggestion
                return self._compare_attr(_ds_id_dict, key, data)
```

Reply:
Not sure this will work here. I only want to return if this is false. If it is tru I want to check the rest before I return true.

Reply:
Ah, ok. Makes sense. Sorry.